### PR TITLE
[DOC] Minor fixes for job documentation

### PR DESCRIPTION
**Pull request by landlord-matt**
## Changes

Simply removing the migration section of `control_run_state` that is no longer needed and makes [the rendered documentation](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/job#control_run_state) look ugly. Also place the tags section last by Terraform documentation convention and fix remaining links.

## Tests

- [ ] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: docs/resources/job.md [hunk 1]
```diff
@@ -82,27 +82,23 @@ The resource supports the following arguments:
...
+* `queue` - (Optional) The queue status for the job. See [queue Configuration Block](#queue-configuration-block) below.
 * `always_running` - (Optional, Deprecated) (Bool) Whenever the job is always running, like a Spark Streaming application, on every update restart the current active run or start it again, if nothing it is not running. False by default. Any job runs are started with `parameters` specified in `spark_jar_task` or `spark_submit_task` or `spark_python_task` or `notebook_task` blocks.
 * `run_as` - (Optional) The user or the service prinicipal the job runs as. See [run_as Configuration Block](#run_as-configuration-block) below.
 * `control_run_state` - (Optional) (Bool) If true, the Databricks provider will stop and start the job as needed to ensure that the active run for the job reflects the deployed configuration. For continuous jobs, the provider respects the `pause_status` by stopping the current active run. This flag cannot be set for non-continuous jobs.
-
-  When migrating from `always_running` to `control_run_state`, set `continuous` as follows:
```
Comment (landlord-matt) [inline docs/resources/job.md:90]: I don't understand what this section is about, it probably isn't relevant for most people today and it looks ugly in the [documentation site](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/job#control_run_state)

File: docs/resources/job.md [hunk 2]
```diff
@@ -294,22 +290,6 @@ This block describes upstream dependencies of a given task. For multiple upstrea
 
 -> **Note** Similar to the tasks themselves, each dependency inside the task need to be declared in alphabetical order with respect to task_key in order to get consistent Terraform diffs.
 
-### tags Configuration Map
```
Comment (landlord-matt) [inline docs/resources/job.md:297]: Tags sections are usually placed last

**Comment by landlord-matt** [comment]
@mgyucht: Any thoughts on this? In my opinion these changes should be uncontroversial and let everyone focus on the actual documentation without being distracted.

**Comment by landlord-matt** [comment]
Perhaps Mr. @nkvuong? Sorry to always bother you.

**Review by tanmay-db — APPROVED** [review]

**Comment by tanmay-db** [comment]
Hi @landlord-matt, to confirm before merging, why is the migration guide no longer needed? I understand that it's deprecated but still supported right? So we should have the migrating guide until it's no longer supported. 

For the time being, I am going ahead with release without this PR since there are some bug fixes that we want to not blocked but will check up on this PR tomorrow for the next release.

**Comment by tanmay-db** [comment]
I will check back up on this PR tomorrow, please feel free to reply whenever comfortable with you.

**Comment by landlord-matt** [comment]
@tanmay-db : Thanks for reviewing. I think this PR will help everyone using Databricks Jobs with Terraform. I reverted the change regarding the migration guide. My original thought was that the downsides of the ugly documentation index for the majority of people using recent versions outweighed the benefits for the minority of users that want to migrate in 2024. Let's merge these uncontroversial changes and separate this change to a separate PR in the future.

**Review by tanmay-db — APPROVED** [review]

**Pull request merged**